### PR TITLE
Create nf-syn28545963-s3.yaml

### PR DESCRIPTION
**Pull request by allaway**
This is to create an S3 bucket for a cloud delivery from NCBI for @ajs3nj.

The bucket needs to have permissions equivalent to this policy:

```
{
  "Id": "Policy1665001491261",
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "NCBIDataDeliveryAccess",
      "Action": "s3:*",
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::my-example-bucket",
        "arn:aws:s3:::my-example-bucket/*"
      ],
      "Principal": {
        "AWS": [
          "arn:aws:iam::184059545989:role/NCBI-CSVM-Service",
          "arn:aws:iam::783971887864:role/NCBI-CSVM-Service"
        ]
      }
    }
  ]
}
```
If we provision a bucket this way, will it have these permissions? It seems like this policy would give all s3:* permissions to the NCBI IAM users.

### CODE REVIEW

File: config/prod/nf-syn28545963-s3.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  # You may omit them if you do not need to override the defaults.
+  # (Optional) true (default) to encrypt bucket, false for no encryption
+  # (Optional) true for read-write bucket, false (default) for read-only bucket
+  AllowWriteBucket: 'true'
+  # (Optional) Synapse username (default: ""), required if AllowWriteBucket=true
+  # (Optional) Allow accounts, groups, and users to access bucket (default is no access).
+  GrantAccess:
+    - 'arn:aws:iam::325565585839:root'   # Required ARN for a synapse bucket
+    - 'arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/robert.allaway@sagebase.org'
+    - 'arn:aws:sts::563295687221:assumed-role/AWSReservedSSO_Developer_baa6fed639faf5e7/sasha.scott@sagebase.org'
```
Comment (zaro0508) [inline config/prod/nf-syn28545963-s3.yaml:22]: we are planning deprecate sandbox and use identity-central account for cross account access instead.  we are just waiting on PR https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/603 to merge.

**Review by zaro0508 — COMMENTED** [review]
looks good in general.  will provide replacements for sage user ARNs shortly.  also the policy for all buckets provisioned this way are like this..
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "SynapseObjectAccess",
      "Effect": "Allow",
      "Principal": {
        "AWS": [
          "arn:aws:iam::325565585839:root",
          "arn:aws:iam::184059545989:role/NCBI-CSVM-Service"
        ]
      },
      "Action": [
        "s3:*Object*",
        "s3:*MultipartUpload*"
      ],
      "Resource": "arn:aws:s3:::your-new-bucket/*",
      "Condition": {
        "StringEquals": {
          "aws:PrincipalAccount": "325565585839"
        }
      }
    },
    {
      "Sid": "BucketAccess",
      "Effect": "Allow",
      "Principal": {
        "AWS": [
          "arn:aws:iam::325565585839:root",
          "arn:aws:iam::184059545989:role/NCBI-CSVM-Service"
        ]
      },
      "Action": [
        "s3:ListBucket*",
        "s3:GetBucketLocation"
      ],
      "Resource": "arn:aws:s3:::your-new-bucket"
    },
    {
      "Sid": "ReadObjectAccess",
      "Effect": "Allow",
      "Principal": {
        "AWS": [
          "arn:aws:iam::325565585839:root",
          "arn:aws:iam::184059545989:role/NCBI-CSVM-Service"
        ]
      },
      "Action": [
        "s3:GetObject",
        "s3:GetObjectAcl",
        "s3:ListMultipartUploadParts"
      ],
      "Resource": "arn:aws:s3:::your-new-bucket/*"
    },
    {
      "Sid": "PutObjectAccess",
      "Effect": "Allow",
      "Principal": {
        "AWS": [
          "arn:aws:iam::325565585839:root",
          "arn:aws:iam::184059545989:role/NCBI-CSVM-Service"
        ]
      },
      "Action": [
        "s3:PutObject",
        "s3:PutObjectAcl",
        "s3:DeleteObject",
        "s3:*MultipartUpload*"
      ],
      "Resource": "arn:aws:s3:::your-new-bucket/*"
    }
  ]
}
```

File: config/prod/nf-syn28545963-s3.yaml (hunk 1 above)
Comment (allaway) [reply config/prod/nf-syn28545963-s3.yaml:22]: Thanks Khai!

Comment (zaro0508) [reply config/prod/nf-syn28545963-s3.yaml:22]: @allaway here are the ARNs you'll need..

```
arn:aws:sts::526515999252:assumed-role/AWSReservedSSO_S3ExternalCollab_40c062f682e7f3f5/robert.allaway@sagebase.org

arn:aws:sts::526515999252:assumed-role/AWSReservedSSO_S3ExternalCollab_40c062f682e7f3f5/sasha.scott@sagebase.org

```

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/prod/nf-syn28545963-s3.yaml (hunk 1 above)
Comment (allaway) [reply config/prod/nf-syn28545963-s3.yaml:22]:
```suggestion
    - 'arn:aws:sts::526515999252:assumed-role/AWSReservedSSO_S3ExternalCollab_40c062f682e7f3f5/robert.allaway@sagebase.org'
    - 'arn:aws:sts::526515999252:assumed-role/AWSReservedSSO_S3ExternalCollab_40c062f682e7f3f5/sasha.scott@sagebase.org'
```

Comment (allaway) [reply config/prod/nf-syn28545963-s3.yaml]: Thanks Khai (again)!

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**